### ISSUE TITLE
Bug on Android 4.4.2

### ISSUE THREAD

**Issue by Rylo4** [mogol/flutter_secure_storage#2]
While trying to run my flutter application on an Android 4.4.2 phone, I receive the following error:

E/FlutterSecureStoragePl(27505): Registration failed
E/FlutterSecureStoragePl(27505): java.lang.NullPointerException: chain == null
E/FlutterSecureStoragePl(27505): 	at java.security.KeyStore$PrivateKeyEntry.<init>(KeyStore.java:1205)
E/FlutterSecureStoragePl(27505): 	at java.security.KeyStoreSpi.engineGetEntry(KeyStoreSpi.java:374)
E/FlutterSecureStoragePl(27505): 	at java.security.KeyStore.getEntry(KeyStore.java:644)
E/FlutterSecureStoragePl(27505): 	at com.it_nomads.fluttersecurestorage.ciphers.StorageCipher18Implementation.createKeysIfNeeded(StorageCipher18Implementation.java:78)
E/FlutterSecureStoragePl(27505): 	at com.it_nomads.fluttersecurestorage.ciphers.StorageCipher18Implementation.<init>(StorageCipher18Implementation.java:29)
E/FlutterSecureStoragePl(27505): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.<init>(FlutterSecureStoragePlugin.java:44)
E/FlutterSecureStoragePl(27505): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.registerWith(FlutterSecureStoragePlugin.java:32)
E/FlutterSecureStoragePl(27505): 	at io.flutter.plugins.GeneratedPluginRegistrant.registerWith(GeneratedPluginRegistrant.java:21)

How can I fix this? It works perfectly on other Android versions.

**Comment by mogol** [comment]
I can't reproduce this issue, but I found similar issue on [StackOverflow](https://stackoverflow.com/questions/23977407/android-4-3-keystore-chain-null-while-trying-to-retrieve-keys)

Could you try this fix - make KEY_ALIAS unique between apps ?
```
public class StorageCipher18Implementation implements StorageCipher {

  private final String KEY_ALIAS;
  private static final String KEYSTORE_PROVIDER_ANDROID = "AndroidKeyStore";
  private static final String TYPE_RSA = "RSA";


  public StorageCipher18Implementation(Context context) throws Exception {
    KEY_ALIAS = context.getPackageName() + ".FlutterSecureStoragePluginKey";
    createKeysIfNeeded(context);
  }

```

**Comment by Rylo4** [comment]
Thanks, this fixed a large part of it, at least for the release builds. It still occurs sometimes at debug mode, but this is fine for me.

**Comment by mogol** [comment]
Great, will release this fix

**Comment by mogol** [comment]
Published fix as 2.0.0.
